include/seastar/coroutine/exception.hh

```diff
@@ -68,7 +68,7 @@ namespace coroutine {
 /// ```
 struct exception {
     std::exception_ptr eptr;
-    explicit exception(std::exception_ptr eptr) : eptr(std::move(eptr)) {}
+    explicit exception(std::exception_ptr&& eptr) noexcept : eptr(std::move(eptr)) {}
```

Comment:
What happens if someone supplies a const exception_ptr? Why does that deserve a compilation failure?